nucleus/rna/views.py

```diff
@@ -55,22 +56,21 @@ def get_queryset(self):
 
 
 @cache_page(RNA_JSON_CACHE_TIME)
+@last_modified(get_last_modified_date)
 @require_safe
 def export_json(request):
-    mod_date = request.GET.get('last-modified')
-    if mod_date == 'all':
+    if request.GET.get('all') == 'true':
         return HttpResponseJSON(models.Release.objects.all_as_list())
 
+    mod_date = request.META.get('HTTP_IF_MODIFIED_SINCE')
     if mod_date:
-        try:
-            # parse_datetime can return None or raise ValueError if value is not parsable
-            mod_date = parse_datetime(mod_date)
-        except ValueError:
-            mod_date = None
+        mod_date = parse_http_date_safe(mod_date)
+        if mod_date:
+            mod_date = datetime.datetime.fromtimestamp(mod_date, datetime.timezone.utc)
 
     if mod_date:
         query = models.Release.objects.recently_modified(mod_date=mod_date)
     else:
-        query = models.Release.objects.recently_modified(days_ago=14)
+        query = models.Release.objects.recently_modified(days_ago=30)
```

Comment:
This works! I's wondering if we can reduce `if mod_date` and I came up with this:

```python
@cache_page(RNA_JSON_CACHE_TIME)
@last_modified(get_last_modified_date)
@require_safe
def export_json(request):
    if request.GET.get('all') == 'true':
        return HttpResponseJSON(models.Release.objects.all_as_list())

    mod_date = (parse_http_date_safe(request.META.get('HTTP_IF_MODIFIED_SINCE')) or
                (datetime.datetime.utcnow() - datetime.timedelta(days=30)).timestamp())
    mod_date = datetime.datetime.fromtimestamp(mod_date, datetime.timezone.utc)

    query = models.Release.objects.recently_modified(mod_date=mod_date)

    return HttpResponseJSON([o.to_dict() for o in query])
```

no need to change your code though, just an idea

Reply:
Hmm... Interesting. The issue is that `parse_http_date_safe` returns an `int` (unix timestamp). I agree though that the current code isn't the most elegant. I'll keep playing with it. Thanks for taking a look!

Reply:
Oh! Nm. I got lost in the parens. I see you accounted for that. Hmmm...

Reply:
I updated things. I left one if statement in there just for readability reasons. I just like it better 🤷‍♂️ but it's much more clean and succinct thanks to your suggestion of just using `mod_date` in both cases instead of `days_ago`. Whaddya think?

Reply:
best solution!